packages/sdk/src/evm/common/ens.ts

```diff
@@ -0,0 +1,55 @@
+import { DEFAULT_API_KEY } from "../../core/constants/urls";
+import { Address, AddressOrEns, AddressOrEnsSchema } from "../schema";
+import { providers } from "ethers";
+
+type CachedEns = {
+  address: string | null;
+  expirationTime: Date;
+};
+
+const ENS_CACHE = new Map<string, CachedEns>();
+
+// TODO: Respect SDK RPC configuration and don't pull straight from ethers
+const provider = new providers.JsonRpcProvider(
+  `https://ethereum.rpc.thirdweb.com/${DEFAULT_API_KEY}`,
+);
+
+async function refreshCache(ens: string): Promise<string | null> {
+  const address = await provider.resolveName(ens);
```

Comment:
can you make sure this works for cb.id and ens with subdomains? like thirdweb.deployer.eth ?